datayoga/blocks/expression.py

```diff
@@ -32,12 +32,41 @@ def search(self, data: Any) -> Any:
         """
         pass
 
+    def test(self, data: Any) -> bool:
+        """Test a where clause for an SQL statement
+
+        Args:
+            data (Any): Data
+
+        Returns:
+            boolean: True if matches, else False
+        """
+        pass
+
 
 class SQLExpression(Expression):
     def compile(self, expression: str):
         self.conn = sqlite3.connect(":memory")
         self.expression = expression
 
+    def test(self, data: Any) -> bool:
+        """Test a where clause for an SQL statement
+
+        Args:
+            data (Any): Data
+
+        Returns:
+            boolean: True if matches, else False
+        """
+        clauses = []
+        for k, v in data.items():
+            clauses.append(f"{json.dumps(v)} as '{k}'")
+
+        from_clause = f"select {','.join(clauses)}"
+        print(f"select 1 from ({from_clause}) where {self.expression}")
+        print(self.conn.execute(f"select 1 from ({from_clause}) where {self.expression}").fetchone())
+        return self.conn.execute(f"select 1 from ({from_clause}) where {self.expression}").fetchone() is not None
```

Comment:
you can use a variable instead of repeating it...

Reply:
refactored